lib/common/options_display.c

```diff
@@ -13,19 +13,197 @@
+                found_default = true;
+                g_string_append(buf, " (default)");
+            }
+        }
+        free(str);
+
+        out->list_item(out, "Allowed values", "%s", buf->str);
+        g_string_free(buf, TRUE);
+
+    } else if (option->default_value != NULL) {
+        out->list_item(out, "Allowed values", "%s (default: \"%s\")",
+                       option->type, option->default_value);
+
+    } else {
+        out->list_item(out, "Allowed values", "%s (default: none)",
+                       option->type);
+    }
+}
+
+/*!
+ * \internal
+ * \brief Output a single option's metadata
+ *
+ * \param[in,out] out     Output object
+ * \param[in]     option  Option to add
+ */
+static void
+add_option_metadata_default(pcmk__output_t *out,
+                            const pcmk__cluster_option_t *option)
+{
+    const char *desc_short = option->description_short;
+    const char *desc_long = option->description_long;
+
+    CRM_ASSERT((desc_short != NULL) || (desc_long != NULL));
+    CRM_ASSERT(option->type != NULL);
+
+    if (desc_short == NULL) {
+        desc_short = desc_long;
+        desc_long = NULL;
+    }
+
+    add_desc_default(out, false, desc_short, option->name);
+
+    out->begin_list(out, NULL, NULL, NULL);
+    add_desc_default(out, true, desc_long, option->name);
+    add_allowed_values_default(out, option);
+    out->end_list(out);
+}
+
+/*!
+ * \internal
+ * \brief Output the metadata for a list of options
+ *
  * \param[in,out] out   Output object
- * \param[in]     tag   Name of element to add (\c PCMK_XE_LONGDESC or
- *                      \c PCMK_XE_SHORTDESC)
- * \param[in]     desc  Textual description to add
+ * \param[in]     args  Message-specific arguments
+ *
+ * \return Standard Pacemaker return code
+ *
+ * \note \p args should contain the following:
+ *       -# Fake resource agent name for the option list (ignored)
+ *       -# Short description of option list
+ *       -# Long description of option list
+ *       -# Filter: If not \c pcmk__opt_context_none, include only those options
+ *          whose \c context field is equal to this filter.
+ *       -# <tt>NULL</tt>-terminated list of options whose metadata to format
+ */
+PCMK__OUTPUT_ARGS("option-list", "const char *", "const char *", "const char *",
+                  "enum pcmk__opt_context", "const pcmk__cluster_option_t *")
+static int
+option_list_default(pcmk__output_t *out, va_list args)
+{
+    const char *name G_GNUC_UNUSED = va_arg(args, const char *);
+    const char *desc_short = va_arg(args, const char *);
+    const char *desc_long = va_arg(args, const char *);
+    enum pcmk__opt_context filter = (enum pcmk__opt_context) va_arg(args, int);
+    const pcmk__cluster_option_t *option_list =
+        va_arg(args, const pcmk__cluster_option_t *);
+
+    CRM_ASSERT((out != NULL) && (desc_short != NULL) && (desc_long != NULL)
+               && (option_list != NULL));
+
+    out->info(out, "%s", desc_short);
```

Comment:
Nowhere. This is the top level -- a short description and long description of the option list.

The XML output has a name (for example, `"cluster-options"`) because that's required by OCF. I don't think it sheds much light on the text output, however, so I dropped the name from default output.

Reply:
Ah, I missed that. It's fine as is